### PR TITLE
fix: refactor cookiebar layout for mobile

### PR DESCRIPTION
**Pull request by rbedeoan-plenty**
[AB#128330](https://dev.azure.com/plentymarkets/0af57df9-8662-4901-bb97-8e88b07ff0c9/_workitems/edit/128330)

### CODE REVIEW

File: apps/web/components/Cookiebar/Cookiebar.vue [hunk 1]
```diff
@@ -1,24 +1,24 @@
 <template>
   <div
     v-if="visible"
-    class="fixed max-sm:flex max-sm:flex-col max-sm:justify-between z-50 w-full xl:w-3/5 xl:right-2 bottom-0 max-md:bottom-[3.9rem] shadow-2xl p-3 bg-white rounded overflow-auto top-0 sm:top-auto"
+    class="fixed cookieBarHeightForMobile max-sm:flex max-sm:flex-col max-sm:justify-between z-50 w-full xl:w-3/5 xl:right-2 bottom-0 shadow-2xl p-3 bg-white rounded overflow-auto end-0 sm:top-auto"
```
Comment (abocsan-plenty) [inline apps/web/components/Cookiebar/Cookiebar.vue]: 1. Putting a fixed height on the whole cookie bar will cause double scroll. We don't need fixed height on the whole cookie bar.

Screenshot 2024-10-01 at 15 45 45

2. Removing `max-md:bottom-[3.9rem]` breaks a functionality we implemented a few days ago for PSI, please don't remove code if you are not 100% sure.

File: apps/web/components/Cookiebar/Cookiebar.vue [hunk 2]
```diff
@@ -111,7 +111,7 @@
       <div class="w-full flex flex-col xl:flex-row mt-5 gap-2 mb-2">
         <div class="flex-1">
           <UiButton
-            class="w-full h-12"
+            class="w-full h-10"
```
Comment (abocsan-plenty) [inline apps/web/components/Cookiebar/Cookiebar.vue]: Could you please confirm that you've checked for PSI errors using both plugins, as outlined in the task requirements ?

Screenshot 2024-10-01 at 15 31 59

File: apps/web/components/Cookiebar/Cookiebar.vue [hunk 3]
```diff
@@ -210,3 +210,10 @@ const getCookiePropertyValue = (cookie: Cookie, propertyKey: string) => {
   return cookie[propertyKey as keyof Cookie]?.toString() || '';
 };
 </script>
+<style>
+@media screen and (max-width: 500px) {
+  .cookieBarHeightForMobile {
+    height: 512.77px;
```
Comment (abocsan-plenty) [inline apps/web/components/Cookiebar/Cookiebar.vue]: While there are rare occasions when we're compelled to write custom CSS due to limitations in Tailwind, this case doesn't fall into that category.

**Review by abocsan-plenty — CHANGES_REQUESTED** [review]

File: apps/web/components/Cookiebar/Cookiebar.vue (hunk 2 above)
Comment (rbedeoan-plenty) [reply apps/web/components/Cookiebar/Cookiebar.vue]: I checked both plugins for psi inconsistency, maybee you can provide more context (pm)
localhost-lighthouse-accesibility
plentyshop-accesibility
wave-localhost
wave-plentymarkets

File: apps/web/components/Cookiebar/Cookiebar.vue (hunk 1 above)
Comment (rbedeoan-plenty) [reply apps/web/components/Cookiebar/Cookiebar.vue]: Thank you for your time
1.From task description this is what i understood.  (maybee you can provide more context (pm)
2.(max-md:bottom-[3.9rem) was not removed accidentaly(maybee you can provide more context (pm)

File: apps/web/components/Cookiebar/Cookiebar.vue [hunk 4]
```diff
@@ -32,7 +32,7 @@
         </template>
       </div>
     </div>
-    <div v-else class="max-sm:h-full overflow-y-auto h-80 pb-2">
+    <div v-else class="sm:h-1/2 max-sm:h-full overflow-y-auto h-80 pb-2">
```
Comment (abocsan-plenty) [inline apps/web/components/Cookiebar/Cookiebar.vue]: leaving h-full for mobile will result in:

image

please test that the scrolling works for further settings as well on mobile.

Comment (abocsan-plenty) [reply apps/web/components/Cookiebar/Cookiebar.vue]: also please check that the desktop version of the cookiebar remains unchanged:

image

**Review by abocsan-plenty — CHANGES_REQUESTED** [review]

File: docs/changelog/changelog_en.md [hunk 5]
```diff
@@ -98,6 +98,7 @@
 
 ### 👷 Changed
 
+- Change the mobile design of the cookie bar layout
```
Comment (abocsan-plenty) [inline docs/changelog/changelog_en.md]: this is old version, please create a `### 👷 Changed` category in the newest version and put the changelog there.

**Review by abocsan-plenty — CHANGES_REQUESTED** [review]

**Review by abocsan-plenty — APPROVED** [review]

**Pull request merged**